src/main/java/network/brightspots/rcv/ResultsWriter.java

```diff
@@ -315,13 +360,15 @@ private void generateSummaryCsv(
   // param: sliceId indicates the specific slice ID we're reporting results for (null means all)
   // param: outputPath is the path to the output file, minus its extension
   // param: extended include additional details in the output file?
-  private void generateSummaryCsvHelper(
+  private void generateSummaryCsv(
           RoundTallies roundTallies,
           List<String> candidateOrder,
-          TabulateBySlice slice,
-          String sliceId,
-          String outputPath,
-          boolean extended) throws IOException {
+          ResultFile resultFile) throws IOException {
+    if (resultFile.resultType != ResultType.SUMMARY_REPORT
+            && resultFile.resultType != ResultType.DETAILED_REPORT) {
+      throw new IllegalArgumentException("ResultFile provided non-CSV type "
```

Comment:
This function builds specific `.csv` reports: the round by round results files that (for now) are either `ResultType.SUMMARY_REPORT` or `ResultType.DETAILED_REPORT`. `RCTAB_CVR` is built in a [different function](https://github.com/BrightSpots/rcv/pull/894/files#diff-50824a8345c807649575fd209dfd6ee673e24eeb8858eff58b150976221a8003R675). If any caller expected this function to build an `RCTAB_CVR` we would have a problem!

Reply:
I have also (1) renamed the function from `generateResultsCsv` to `generateCsvReport` (https://github.com/BrightSpots/rcv/pull/894/commits/33bd30946f101c1894f903b3feb20efc463fa5ca#diff-50824a8345c807649575fd209dfd6ee673e24eeb8858eff58b150976221a8003R662) and (2) updated the comment to say "non-CSV Report Type" instead of just "non-CSV type"